src/o365/spfx/commands/project/project-externalize/rules/PnPJsRule.ts

```diff
@@ -0,0 +1,22 @@
+import { BasicDependencyRule } from "./BasicDependencyRule";
```

Comment:
Why do we need a specific rule for PnPjs?

Reply:
Pnpjs is special given the fact you can either use the bundle or separate librairies. I anticipate that we'll probably have feedback around that and will need to implement something specific.

Reply:
I'm not sure if you have experience with using PnPjs from externals, but I wonder if we won't run into problems with setting up the correct dependency chain so that PnPjs modules are loaded in the right order.

Reply:
well there are two cases:
- using the bundle, it's fairly straightforward (@pnp/pnpjs)
- using individual packages, it can quickly become a mess depending on which individual package they are using and making sure we're respecting the order.

For the second case, we probably need to build a dependency tree and suggest the ones we detect + parents (with dedupe), adding all the dependencies following the tree for each package....

Again I created the rule knowing we'll probably need it at some point but I haven't implemented it so far.

https://pnp.github.io/pnpjs/documentation/deployment/

What are your thoughts?
Should we rule it as an edge case and tell people to move to the bundle? Should we implement it for initial release of this command? Any other option?

Reply:
I went ahead with my dependency tree implementation in the rule. closing this comment